test/e2e/operator_test.go

```diff
@@ -572,50 +578,51 @@ func TestDNSOverTLSForwarding(t *testing.T) {
 		t.Fatalf("version %s not found for clusteroperator %s", statuscontroller.OpenshiftCLIVersionName, opName)
 	}
 
-	// Create the upstream resolver Pods.
-	upstreamResolver1 := upstreamTLSPod(tlsUpstreamName+"-1", tlsUpstreamNamespace.Name, coreImage, upstreamTLSConfigMap)
-	if err := cl.Create(context.TODO(), upstreamResolver1); err != nil {
-		t.Fatalf("failed to create pod %s/%s: %v", upstreamResolver1.Namespace, upstreamResolver1.Name, err)
+	// Create the upstream resolver Pods and the client pod
+	upstreamResolver := upstreamTLSPod(tlsUpstreamName, tlsUpstreamNamespace.Name, coreImage, upstreamTLSConfigMap)
+	if err := cl.Create(context.TODO(), upstreamResolver); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", upstreamResolver.Namespace, upstreamResolver.Name, err)
 	}
-	upstreamResolver2 := upstreamTLSPod(tlsUpstreamName+"-2", tlsUpstreamNamespace.Name, coreImage, upstreamTLSConfigMap)
-	if err := cl.Create(context.TODO(), upstreamResolver2); err != nil {
-		t.Fatalf("failed to create pod %s/%s: %v", upstreamResolver2.Namespace, upstreamResolver2.Name, err)
+	testClient := buildPod("test-client-tls", tlsUpstreamName, cliImage, []string{"sleep", "3600"})
```

Comment:
So I assume the TLDR is:
```
// This looks in both upstream resolvers because the forwarding policy could be random. This also serves the purpose
// of testing that one server certificate will work for multiple upstreams using the same ServerName.
```

We feel not necessary to test multiple upstream with one certificate? Is it a bit orthogonal? (Sorry I think you Miciah talked about this today, but I missed it...)

Reply:
Exactly. We're testing that we can use the CA to validate the server cert of the upstream and trusting that CoreDNS will move on to the next upstream if the first one doesn't work.